apps/src/code-studio/header.js

```diff
@@ -107,7 +107,9 @@ header.build = function (
 
   // Set up a navigation handler, in case we contain levels that don't
   // require a page reload when switching between them.
-  setupNavigationHandler(lessonData);
+  if (lessonData.levels.some(level => level.uses_lab2)) {
```

Comment:
Added a change to only setup the browser navigation handler for progressions with lab2 levels. This is because a few UI tests were failing due to the "levelId" being added to the browser history stack.

For context: the UI test step for "I get redirected to" [checks that the state is `null`](https://github.com/code-dot-org/code-dot-org/blob/0d41b2aa59eb58abcd6ddc9ef092ae9e1e9d9e5f/dashboard/test/ui/features/step_definitions/steps.rb#L1317), but I feel like that may not be necessary? In any case though, I figured this would be an easier fix, rather than reworking core UI test steps for now.